src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
I disagree. Shouldn't take very long to implement at all and will prevent us from ever experiencing future issues.